icicle/include/hash/hash.cuh

```diff
@@ -123,6 +126,17 @@ namespace hash {
       Image* output,
       const device_context::DeviceContext& ctx) const = 0;
 
+    virtual cudaError_t absorb_2d(
+      const Matrix<PreImage>* inputs,
+      Image* states,
+      unsigned int number_of_inputs,
+      uint64_t number_of_rows,
```

Comment:
what is number_of_rows? of what? Matrices have [width, height] fields so it's not clear to me what this refers to.